public/components/map_top_nav/top_nav_menu.tsx

```diff
@@ -99,26 +97,22 @@ export const MapTopNavMenu = ({
 
   const handleQuerySubmit = ({ query, dateRange }: { query?: Query; dateRange: TimeRange }) => {
     setIsUpdatingLayerRender(true);
-    if (query) {
-      setMapState({ ...mapState, query });
-    }
-    if (dateRange) {
-      setMapState({ ...mapState, timeRange: dateRange });
+    if (query || dateRange) {
+      setMapState({
+        ...mapState,
+        ...(query && { query }),
+        ...(dateRange && { timeRange: dateRange }),
+      });
```

Comment:
can you split into two if and update at end?

Reply:
If both if statements are true, the second call to setMapState will overwrite the changes made by the first call to setMapState.So that we need combine the two setMapState calls into one,

Reply:
From above it looks like dateRange cannot be undefined. Do we need a check there? create a local const and update the value if query or range is true, and , at end, set those local const in setMapState.

Reply:
updated.